pyccel/codegen/compiling/project.py

```diff
@@ -0,0 +1,327 @@
+from pathlib import Path
+from pyccel.errors.errors  import Errors
+
+errors = Errors()
+
+class CompileTarget:
+    """
+    Class describing a compilation target.
+
+    Class describing the compilation target of a translated Python file.
+    The class contains all the information necessary to create the
+    necessary targets in a build system (e.g. CMake, meson).
+
+    Parameters
+    ----------
+    name : str
+        The unique identifier for the target.
+    pyfile : Path
+        The absolute path to the Python file that was translated.
+    file : str | Path
+        The absolute path to the low-level translation of the Python file.
+    wrapper_files : dict[Path, iterable[str]]
+        A dictionary whose keys are the absolute paths to the generated wrapper files,
+        and whose values are iterables containing the names of the stdlib targets for
+        these additional files.
+    program_file : str | Path, optional
+        The absolute path to the low-level translation of the program found
+        in the Python file (if the file contained a program).
+    stdlib_deps : iterable[str]
+        An iterable containing the names of the stdlib targets of this object.
+    """
+    __slots__ = ('_name', '_pyfile', '_file', '_wrapper_files',
+                 '_program_file', '_dependencies', '_stdlib_deps',
+                 '_wrapper_stdlib_deps')
+    def __init__(self, name, pyfile, file, wrapper_files, program_file, stdlib_deps):
+        self._name = name
+        self._pyfile = pyfile
+        self._file = Path(file)
+        self._wrapper_files = wrapper_files
+        self._program_file = Path(program_file) if program_file is not None else program_file
+        self._dependencies = []
+        self._stdlib_deps = list(stdlib_deps)
+
+    @property
+    def name(self):
+        """
+        The unique identifier for the target.
+
+        The unique identifier for the target.
+        """
+        return self._name
+
+    @property
+    def pyfile(self):
+        """
+        The absolute path to the Python file that was translated.
+
+        The absolute path to the Python file that was translated.
+        """
+        return self._pyfile
+
+    @property
+    def file(self):
+        """
+        The absolute path to the low-level translation of the Python file.
+
+        The absolute path to the low-level translation of the Python file.
+        """
+        return self._file
+
+    @property
+    def wrapper_files(self):
+        """
+        The absolute path to the generated wrapper files.
+
+        The absolute path to the generated wrapper files.
+        """
+        return self._wrapper_files
+
+    @property
+    def program_file(self):
+        """
+        The absolute path to the low-level translation of the program.
+
+        The absolute path to the low-level translation of the program found
+        in the Python file (if the file contained a program).
+        """
+        return self._program_file
```

Comment:
Maybe mention that `None` is returned if the Python file did not contain a program?